packages/server/lib/controllers/sync/postTrigger.ts

```diff
@@ -0,0 +1,102 @@
+import { z } from 'zod';
+
+import { logContextGetter } from '@nangohq/logs';
+import { records as recordsService } from '@nangohq/records';
+import { SyncCommand, errorManager, syncManager } from '@nangohq/shared';
+import { requireEmptyQuery, zodErrorToHTTP } from '@nangohq/utils';
+
+import { asyncWrapper } from '../../utils/asyncWrapper.js';
+import { getOrchestrator } from '../../utils/utils.js';
+
+import type { PostTrigger } from '@nangohq/types';
+
+const bodyValidation = z
+    .object({
+        syncs: z.array(
+            z.union([z.string(), z.object({ name: z.string(), variant: z.string() })], {
+                errorMap: () => ({ message: 'Each sync must be either a string or a { name: string, variant: string } object' })
+            })
+        ),
+        full_resync: z.boolean(),
+        connection_id: z.string().optional(),
+        provider_config_key: z.string().optional()
+    })
+    .strict();
+
+const headersValidation = z.object({
+    'Provider-Config-Key': z.string().optional(),
+    'Connection-Id': z.string().optional()
+});
+
+const orchestrator = getOrchestrator();
+
+export const postPublicTrigger = asyncWrapper<PostTrigger>(async (req, res) => {
+    const emptyQuery = requireEmptyQuery(req);
+    if (emptyQuery) {
+        res.status(400).send({ error: { code: 'invalid_query_params', errors: zodErrorToHTTP(emptyQuery.error) } });
+        return;
+    }
+
+    const body = bodyValidation.safeParse(req.body);
+    if (!body.success) {
+        res.status(400).send({ error: { code: 'invalid_body', errors: zodErrorToHTTP(body.error) } });
+        return;
+    }
+
+    const headers = headersValidation.safeParse(req.headers);
+    if (!headers.success) {
+        res.status(400).send({ error: { code: 'invalid_headers', errors: zodErrorToHTTP(headers.error) } });
+        return;
+    }
+
+    const { syncs, full_resync } = body.data;
```

Comment:
can you reassign just to counter type? this is to prevent desync between zod and types
```suggestion
    const { syncs, full_resync }: PostTrigger["Body"] = body.data;
```